crates/agent-sql/src/evolutions.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{publications::RoleGrant, Capability, CatalogType, Id, TextJson as Json};
+use chrono::prelude::*;
+use serde::Serialize;
+use serde_json::value::RawValue;
+use sqlx::types::Uuid;
+
+// Row is the dequeued task of an evolution operation.
+#[derive(Debug)]
+pub struct Row {
+    pub id: Id,
+    pub created_at: DateTime<Utc>,
+    pub detail: Option<String>,
+    pub draft_id: Id,
+    pub logs_token: Uuid,
+    pub updated_at: DateTime<Utc>,
+    pub user_id: Uuid,
+    pub collections: Json<Box<RawValue>>,
+}
+
+pub async fn dequeue(txn: &mut sqlx::Transaction<'_, sqlx::Postgres>) -> sqlx::Result<Option<Row>> {
+    sqlx::query_as!(
+        Row,
+        r#"select
+            id as "id: Id",
+            created_at,
+            detail,
+            draft_id as "draft_id: Id",
+            logs_token,
+            updated_at,
+            user_id,
+            collections as "collections: Json<Box<RawValue>>"
+        from evolutions where job_status->>'type' = 'queued'
+        order by id asc
+        limit 1
+        for update of evolutions skip locked;
+        "#
+    )
+    .fetch_optional(txn)
+    .await
+}
+
+pub async fn resolve<S>(
+    id: Id,
+    status: &S,
+    txn: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+) -> sqlx::Result<()>
+where
+    S: Serialize + Send + Sync,
+{
+    sqlx::query!(
+        r#"update evolutions set
+            job_status = $2,
+            updated_at = clock_timestamp()
+        where id = $1
+        returning 1 as "must_exist";
+        "#,
+        id as Id,
+        Json(status) as Json<&S>,
+    )
+    .fetch_one(txn)
+    .await?;
+
+    Ok(())
+}
+
+pub struct LiveSpecRow {
+    pub id: Id,
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+    pub last_pub_id: Id,
+    pub catalog_name: String,
+    pub spec_type: Option<CatalogType>,
+    pub spec: Option<Json<Box<RawValue>>>,
+    pub spec_capabilities: Json<Vec<RoleGrant>>,
+    pub user_capability: Option<Capability>,
+}
+
+pub async fn fetch_live_specs(
```

Comment:
Adding this comment later, after having otherwise finished the review:

I completely missed where this is called. You're fetching draft specs, sure. You use the expand-graph function to fetch connected captures / materializations. Where does this fit in?

Reply:
whoops, yea this was vestigial